### ISSUE TITLE
[no-unused-vars] Class constructor private member false positive

### ISSUE THREAD

**Issue by cdalexndr** [typescript-eslint/typescript-eslint#1679]
**Repro**

```JSON
env:
  browser: true
  es6: true
extends:
  - 'eslint:recommended'
  - 'plugin:@typescript-eslint/eslint-recommended'
globals:
  Atomics: readonly
  SharedArrayBuffer: readonly
parser: '@typescript-eslint/parser'
parserOptions:
  ecmaVersion: 2018
  sourceType: module
plugins:
  - '@typescript-eslint'
rules: {}

```

```TS
export class SomeClass{
    constructor(private element: string) {
    }
    
    get():string{
        return this.element;
    }
}
```

**Expected Result**
No error.

**Actual Result**
`error  'element' is defined but never used            no-unused-vars`

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.22.0` |
| `@typescript-eslint/parser`        | `2.22.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `10.16.0` |
| `npm`                              | `6.13.1` |

**Comment by cdalexndr** [comment]
Workaround: move constructor argument to class member:
```
export class SomeClass{
    private element: string;

    constructor(element: string) {
        this.element = element;
    }
}
```

**Comment by bradzacher** [comment]
```
error 'element' is defined but never used no-unused-vars
                                          ^^^^^^^^^^^^^^
```

Make sure you're using our version of no-unused-vars, which specifically handles this case.

**Comment by cdalexndr** [comment]
My configuration file was generated with `eslint --init` wizard where I enabled typescript and installed additional dependencies. 
Shouldn't initialization use the correct version of `no-unused-vars`?
Should I open a new bug for the init issue?

**Comment by bradzacher** [comment]
Not if they haven't configured it to.
They went for the minimal config route, so it does not.
